### PR TITLE
fix: add validation for ALB access log bucket when KMS key is provided

### PR DESCRIPTION
**Pull request by shikha372**
### Issue # (if applicable)

Closes #22031, #21947.

### Reason for this change

Adds a validation with correct error indicating ALB Access log bucket does not support KMS encryption

### Description of changes

Currently access logs bucket encryption with KMS is not supported in case of ALB but while deploying it throws an error indicating the failure with bucket permissions. 
This validation introduces an upfront check to throw an error if `bucket.encryptionKey `is defined.
Documentation: https://docs.aws.amazon.com/elasticloadbalancing/latest/application/enable-access-logging.html

### Description of how you validated changes

Added unit tests for validation.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/alb/application-load-balancer.ts [hunk 1]
```diff
@@ -151,6 +154,66 @@ export class ApplicationLoadBalancer extends BaseLoadBalancer implements IApplic
...
+  public logAccessLogs(bucket: s3.IBucket, prefix?: string) {
+
+    /**
+    * KMS key encryption is not supported on Access Log bucket for ALB, the bucket must use Amazon S3-managed keys (SSE-S3).
+    * See https://docs.aws.amazon.com/elasticloadbalancing/latest/application/enable-access-logging.html#bucket-permissions-troubleshooting
+    */
+
+    if (bucket.encryptionKey) {
+      throw new Error('Encryption key detected. Bucket encryption using KMS keys is unsupported');
+    }
```
Comment (GavinZZ) [inline packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/alb/application-load-balancer.ts:172]: Is this the only different in the `logAccessLogs` call for base load balancer vs application load balancer class?

File: packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/alb/application-load-balancer.ts [hunk 2]
```diff
@@ -151,6 +154,66 @@ export class ApplicationLoadBalancer extends BaseLoadBalancer implements IApplic
...
+   * Enable access logging for this load balancer.
+   *
+   * A region must be specified on the stack containing the load balancer; you cannot enable logging on
+   * environment-agnostic stacks. See https://docs.aws.amazon.com/cdk/latest/guide/environments.html
+   */
+  public logAccessLogs(bucket: s3.IBucket, prefix?: string) {
+
+    /**
+    * KMS key encryption is not supported on Access Log bucket for ALB, the bucket must use Amazon S3-managed keys (SSE-S3).
+    * See https://docs.aws.amazon.com/elasticloadbalancing/latest/application/enable-access-logging.html#bucket-permissions-troubleshooting
```
Comment (GavinZZ) [inline packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/alb/application-load-balancer.ts:167]: Can we also add this to `README.md`

File: packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/alb/application-load-balancer.ts (hunk 1 above)
Comment (shikha372) [reply packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/alb/application-load-balancer.ts:172]: yes only this part to check whether encryption key is enabled

File: packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/alb/application-load-balancer.ts (hunk 2 above)
Comment (shikha372) [reply packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/alb/application-load-balancer.ts:167]: Didn't find any prior section to enable access logging on ALB in Readme, added it with above doc reference.

**Review by GavinZZ — APPROVED** [review]

**Comment by shikha372** [comment]
@Mergifyio update

**Comment by shikha372** [comment]
@Mergifyio update

**Pull request merged**